README.md

```diff
@@ -182,54 +153,78 @@ config/infra/aws/config.json
-          "name": "prod",
-          "type": "profile",
-          "config": {
-            "profile": "prod",
-            "awsDefaultRegion": "us-west-2"
-          }
-        }
-      ]
-    }`
-
-### Credentials in Environment Variables
+      "name": "dev",
+      "type": "profile",
+      "config": {
+        "profile": "default",
+        "awsDefaultRegion": "us-west-2"
+      }
+    },
+    {
+      "name": "prod",
+      "type": "profile",
+      "config": {
+        "profile": "prod",
+        "awsDefaultRegion": "us-west-2"
+      }
+    }
+  ]
+}
+```
 
-Goldstack can read AWS *Access Key ID* and *Secret Access Key* from environment variables. The easiest way is to set the following environment variables:
+Note that Goldstack also supports overriding the path of the default AWS configuration and credentials files:
 
-```bash
-AWS_USER_NAME: [Your user name]
-AWS_ACCESS_KEY_ID: [Your access key id]
-AWS_SECRET_ACCESS_KEY: [Your secret access key]
-AWS_DEFAULT_REGION: [User region]
+```json
+{
+  "users": [
+    {
+      "name": "prod",
+      "type": "profile",
+      "config": {
+        "profile": "prod",
+        "awsDefaultRegion": "us-west-2",
+        "awsConfigFile": "/path/to/config/file",
+        "awsCredentialsFile": "/path/to/credentials/file"
+      }
+    }
+  ]
+}
 ```
 
-The `AWS_USER_NAME` variable is optional but can be useful for explicitly referencing the correct Goldstack user in deployments. The above setup is particularly useful for CI/CD environments. For instance, when using [GitHub Actions](https://github.com/actions), environment variables could be configured as follows:
+### Using process credentials
 
-```yaml
-- name: Deploy UI
-  run: |
-    yarn workspace my-ui deploy dev
-  env:
-    AWS_USER_NAME: dev-user
-    AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
-    AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
-    AWS_DEFAULT_REGION: us-west-2
+There are a number of issues when trying to work with multiple profiles and SSO credentials, see [aws/aws-cli#4982 (comment)](https://github.com/aws/aws-cli/issues/4982#issuecomment-939348934) and [goldstack/goldstack#17](https://github.com/goldstack/goldstack/issues/17).
+
+An excellent way to deal with situations where we do not want to provide the user credentials directly, is to use [process credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html).
+
+Goldstack supports defining process credentials in the `config/infra/aws/config.json` file.
+
+```json
+"users": [
+    {
+      "name": "dev-user",
+      "type": "profile",
+      "config": {
+        "profile": "with-process",
+        "awsDefaultRegion": "us-west-2",
+        "credentialsSource": "process"
+      }
+    }
+  ]
 ```
 
-Where the values of the environment variables are defined in [GitHub Secrets](https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets).
+This will require a `~/.aws/config` file as follows:
```

Comment:
Added to the documentation that the file path can be changed. https://github.com/goldstack/goldstack/pull/102/files#diff-78f54ba2c3b35a3f26d72bde4fe00c756b46b3a00b454b9c670b2be7892e7e37R49 

I think in addition to this would probably good to fix it so that it works out of the box. Does this make change for this comment?

Reply:
> I think in addition to this would probably good to fix it so that it works out of the box. Does this make change for this comment?

If you mean that `"credentialsSource": "process"` should work without `"awsConfigFileName": "/path/to/config/file"` then yes, agreed. I think the code change (`process.env.AWS_SDK_LOAD_CONFIG = '1'`) below should solve this though?